packages/mui-material/src/useMediaQuery/useMediaQuery.test.js

```diff
@@ -291,14 +299,16 @@ describe('useMediaQuery', () => {
         query: PropTypes.string.isRequired,
       };
 
-      render(<Test query="(min-width:2000px)" />);
+      render(<Test query={query} />);
+
       expect(getRenderCountRef.current()).to.equal(1);
       expect(screen.getByTestId('matches').textContent).to.equal('false');
 
-      act(() => {
-        matchMediaInstances[matchMediaInstances.length - 1].instance.matches = true;
-        matchMediaInstances[matchMediaInstances.length - 1].listeners[0]();
+      await act(async () => {
+        matchMediaInstances.get(query).instance.matches = true;
+        matchMediaInstances.get(query).listeners[0]();
```

Comment:
React 19 has to use `await` right?

Reply:
Yes, React recommends using `await act(async () => {})`. Not only in React 19, it's been like this since React 18 AFAIK.

> We recommend using act with await and an async function. Although the sync version works in many cases, it doesn’t work in all cases and due to the way React schedules updates internally, it’s difficult to predict when you can use the sync version.
> 
> We will deprecate and remove the sync version in the future.

Source: https://react.dev/reference/react/act#await-act-async-actfn